### PR TITLE
CPOL-141 Remove the need for reindexing everything at start. Use RocksDB

### PR DESCRIPTION
**Pull request by burmanm**
Migrate the data storage to using RocksDB

### CODE REVIEW

**Comment by pilhuhn** [comment]
I get some confusing messages: 

```
2020-03-26 10:09:40,315 ERROR [com.red.clo.pol.eng.act.QuarkusActionPluginRegister] (main) CPOL270005: Plugin [email] cannot be registered into the engine. Error: [org.hawkular.alerts.api.exception.FoundException: ActionPlugin-email]
2020-03-26 10:09:40,315 INFO  [com.red.clo.pol.eng.act.QuarkusActionPluginRegister] (main) Registered plugin email implemented by class com.redhat.cloud.policies.engine.actions.plugins.EmailActionPluginListener
2020-03-26 10:09:40,320 ERROR [com.red.clo.pol.eng.act.QuarkusActionPluginRegister] (main) CPOL270005: Plugin [webhook] cannot be registered into the engine. Error: [org.hawkular.alerts.api.exception.FoundException: ActionPlugin-webhook]
2020-03-26 10:09:40,320 INFO  [com.red.clo.pol.eng.act.QuarkusActionPluginRegister] (main) Registered plugin webhook implemented by class com.redhat.cloud.policies.engine.actions.plugins.WebhookActionPluginListener
```

**Comment by pilhuhn** [comment]
And then for requests with a non-existing trigger: 

`2020-03-26 10:20:10,597 ERROR [org.haw.ale.eng.imp.isp.IspnDefinitionsServiceImpl] (vert.x-worker-thread-1) CPOL220008: Database Exception. Msg: [Trigger-1234-bd0ee2ec-eec0-44a6-8bb1-29c4179fc21c]`

Is ERROR logging appropriate?

File: engine/src/test/resources/META-INF/microprofile-config.properties [hunk 1]
```diff
@@ -4,6 +4,9 @@ engine.receiver.store-events=true
 # == Infinispan properties
 engine.backend.ispn.reindex=true
 
+engine.backend.ispn.alerts-lifespan=1
```
Comment (pilhuhn) [inline engine/src/test/resources/META-INF/microprofile-config.properties]: Would be good to have a comment that tells the unit

**Comment by burmanm** [comment]
The registering error would happen if the system isn't cleanly shutdown and index was corrupted. In that case one needs to reindex.

**Pull request merged**